File: test/legofix/generics.java
// FIX #1 repro (Java): a class extending a GENERIC base. The base clause hands back a
// `generic_type` node whose text is `Base<String>` — the `<...>` type arguments must be
// stripped so the recorded base name is `Base` and `--lego=Base` surfaces the implementor D.
// (Single-language type names — Base/D — unique to this file so no cross-fixture collision.)
public class Base<T>
{
    public T get() { return null; }
}

class D extends Base<String>
{
    public String get() { return ""; }
}

File: test/legofix/generics.rs
// FIX #1 repro (Rust): a GENERIC impl of a trait for a generic type. The impl's `type:` field
// is `Wrapper<T>` — the `<...>` type arguments must be stripped so the DERIVED name stashed in
// `qualifier` is `Wrapper` and resolves by name, making `--lego=Draw` surface Wrapper.
// (Trait/type names Draw/Wrapper are unique to this file.)
trait Draw
{
    fn draw( &self );
}

struct Wrapper<T>
{
    inner: T,
}

impl<T> Draw for Wrapper<T>
{
    fn draw( &self ) {}
}
